operator/iamkeycontroller/webhook.go

```diff
@@ -23,6 +23,10 @@ func (v *IAMKeyValidator) ValidateCreate(_ context.Context, obj runtime.Object)
 		return fmt.Errorf("an IAMKey named %q should have at least 1 allowed bucket",
 			iamKey.Name)
 	}
+	if iamKey.Spec.WriteConnectionSecretToReference == nil {
+		return fmt.Errorf("an IAMKey named %q should have a connection secret reference",
+			iamKey.Name)
+	}
```

Comment:
This (and the following if's) can be simplified:
```suggestion
	secretRef := iamKey.Spec.WriteConnectionSecretToReference
	if secretRef == nil || secretRef.Name == "" || secretRef.Namespace == "" {
		return fmt.Errorf("an IAMKey named %q requires a connection secret reference with name and namespace",
			iamKey.Name)
	}
```

Reply:
yep good point.